### DISCUSSION TITLE
Possible to apply Mocha test timeouts across a whole Feature? Or Globally?

### DISCUSSION THREAD

**Discussion by ghowardzoonou** [badeball/cypress-cucumber-preprocessor#1251] [Q&A] [answered]
Hello, as per subject - is it possible to apply Mocha test timeouts across a whole Feature? e.g. as per Suites here: [https://mochajs.org/#timeouts](https://mochajs.org/#timeouts)? Or even Globally?

I want individual Scenarios to fail if they are running for >10 mins each, etc.

**Comment by badeball** [comment]
Does this work in a plain Cypress test?

**Reply by ghowardzoonou** [reply, marked as answer]
Hi, so yeh ended going down a bit of a rabbit hole with this one. In the end the way I could get a global test timeout working was by following this approach:

https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/fundamentals__timeout

https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/fundamentals__timeout/cypress/e2e/all-tests-spec.cy.js

**Reply by badeball** [reply]
This is pretty much what I was expecting (ref. https://github.com/cypress-io/cypress/issues/9009). Thus, there's no reason for this library to offer a way to invoke `this.timeout(..)` when it wouldn't make a difference anyway.
